specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json

```diff
@@ -15907,6 +15912,7 @@
         "running",
         "starting",
         "startTaskFailed",
+        "leavingPool",
```

Comment:
Did you already ship client SDK with the current api version? Will they still work without this parameter being sent to server?

Reply:
@hovsepm This is server response. Not request.

Reply:
@xingwu1 why do you mark server response fields as required then? IMHO they should be marked as `"readOnly": true` to prevent client from modifying them. Required field means that those parameters should be present in the request and not response  - https://swagger.io/docs/specification/describing-parameters/

Reply:
readOnly and required field can't co-exist. However, without required field, all the int properties will convert to Nullable<int> which make user difficult to use. So right now, we will keep the required field right now. We will re-consider readonly field.